### ISSUE TITLE
[dataclasses] Equality operator for I3FilterResult and I3SuperDST (Trac #2130)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2122]
I am trying to verify that processing raw to superdst gives the same result on SL6 and SL7, by reading the files in parallel and comparing frame-object by frame-object.

The QFilterMask and I3SuperDST objects always were !=, though when I compared the contents they were the same.

Migrated from https://code.icecube.wisc.edu/ticket/2130
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:15:13", 
    "description": "I am trying to verify that processing raw to superdst gives the same result on SL6 and SL7, by reading the files in parallel and comparing frame-object by frame-object.\n\nThe QFilterMask and I3SuperDST objects always were !=, though when I compared the contents they were the same.", 
    "reporter": "jbellinger", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067313248429", 
    "component": "combo core", 
    "summary": "[dataclasses] Equality operator for I3FilterResult and I3SuperDST", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2018-01-17T21:41:38", 
    "milestone": "", 
    "owner": "jbraun", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20180117T21:46:35: 
* david.schultz commented:

> The actual class of the QFilterMask is `I3FilterResult`.


* david.schultz changed type from "defect" to "enhancement"
* david.schultz changed summary from "Equality operator wanted for QFilterMask and I3SuperDST" to "[dataclasses] Equality operator for I3FilterResult and I3SuperDST"
* david.schultz changed milestone from "" to "IceRec 5.3"
* david.schultz changed component from "analysis" to "icecube offline"

**Comment by icecube-trac** [comment]
Trac update at 20180613T17:49:16: 
* olivas changed milestone from "IceRec 5.3" to "Summer Solstice 2018"
* olivas changed status from "new" to "assigned"
* olivas changed owner from "" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20180626T16:45:17: 
* olivas changed owner from "olivas" to "jbraun"
* olivas commented:

> Jim offered to take this one.  Thanks!

**Comment by icecube-trac** [comment]
Trac update at 20180626T20:06:24: 
* jbraun changed status from "assigned" to "closed"
* jbraun changed resolution from "" to "fixed"
* jbraun commented:

> In changeset 163733:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="163733"
> Add equality operators for I3SuperDST and I3FilterResult.  Closes #2122
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:15:13: 
* olivas commented:

> Milestone Summer Solstice 2018 deleted


* olivas changed milestone from "Summer Solstice 2018" to ""
